include/ddc/detail/dual_discretization.hpp

```diff
@@ -59,6 +61,10 @@ class DualDiscretization
         else if constexpr (std::is_same_v<MemorySpace, Kokkos::HIPSpace>) {
             return m_device_on_host;
         }
+#elif defined(KOKKOS_ENABLE_SYCL)
+        else if constexpr (std::is_same_v<MemorySpace, Kokkos::SYCLDeviceUSMSpace>) {
+            return m_device_on_host;
+        }
 #endif
         else {
             static_assert(!std::is_same_v<MemorySpace, MemorySpace>);
```

Comment:
Is this line meaningful? @tpadioleau

Reply:
Should be, ideally i would like to put false but the compiler will then stop compiling even if the branch is not chosen